### ISSUE TITLE
poor performance using my own robot after finetuning

### ISSUE THREAD

**Issue by zhangwencong317** [Physical-Intelligence/openpi#433]
Recently, I fine-tuned the pi0_base model using data I collected from my own robot(not any of the robots mentioned in the README), and the loss dropped to around 0.0015(30000 steps). However, when I applied the trained weights to my actual robot, the performance was poor.

My application logic involves taking the first ten actions from the model’s inference output for the robot to execute. After completing those actions, I use the current state and image to obtain the next ten actions from the model output, repeating this process to accomplish the task.

Could the issue lie in the training phase or in the logic of applying the model’s inference during deployment?

**Comment by zhangwencong317** [comment]
@dataclasses.dataclass(frozen=True)
class LeRobotAirbotDataConfig(DataConfigFactory):
    # Airbot DataConfig
    @override
    def create(self, assets_dirs: pathlib.Path, model_config: _model.BaseModelConfig) -> DataConfig:

        repack_transform = _transforms.Group(
            inputs=[
                _transforms.RepackTransform(
                    {
                        "top_image": "top_image",       
                        "wrist_image": "wrist_image",     
                        "state": "state",              
                        "actions": "actions",   
                        "task": "prompt",          
                    }
                )
            ]
        )

        data_transforms = _transforms.Group(
            inputs=[airbot_policy.AirbotInputs(action_dim=model_config.action_dim, model_type=model_config.model_type)],
            outputs=[airbot_policy.AirbotOutputs()],
        )
        
        delta_action_mask = _transforms.make_bool_mask(6, -1)
        data_transforms = data_transforms.push(
            inputs=[_transforms.DeltaActions(delta_action_mask)],
            outputs=[_transforms.AbsoluteActions(delta_action_mask)],
        )

        # Model transforms include things like tokenizing the prompt and action targets
        # You do not need to change anything here for your own dataset.
        model_transforms = ModelTransformFactory()(model_config)

        # We return all data transforms for training and inference. No need to change anything here.
        return dataclasses.replace(
            self.create_base_config(assets_dirs),
            repack_transforms=repack_transform,
            data_transforms=data_transforms,
            model_transforms=model_transforms,
        )


    TrainConfig(
        name="pi0_airbot",
        model=pi0.Pi0Config(),
     
        data=LeRobotAirbotDataConfig(
            repo_id="transform2lerobot_data/airbot",
            base_config=DataConfig(
                local_files_only=True, 
                prompt_from_task=True, # prompt_from_task: 
            ),
        ),

        weight_loader=weight_loaders.CheckpointWeightLoader("s3://openpi-assets/checkpoints/pi0_base/params"),

        num_train_steps=30_000,
    ),

**Comment by lidian0428** [comment]
Hi. are there any challenges observed during the training/inference phase? Do the final performance good？

**Comment by meijie-jesse** [comment]
@drew7k4 Hi, so have you find the solutions? I met the same poor performance using my own robot.

**Comment by tahsinkose** [comment]
@drew7k4 similar here. How did you fix it? What was your finetuning dataset size and duration?
